packages/server/database/migrations/20210527170505-addAdditionalRetroTemplates.ts

```diff
@@ -0,0 +1,199 @@
+  'Starfish ⭐',
+  'What Went Well'
+]
+
+const promptsInfo = [
+  {
+    question: 'Drop',
+    description: 'What should we stop?',
+    groupColor: '#52CC52',
+    templateId: 'dropAddKeepImproveDAKITemplate',
+    sortOrder: 0
+  },
+  {
+    question: 'Add',
+    description: 'What should we try or adopt?',
+    groupColor: '#E55C5C',
+    templateId: 'dropAddKeepImproveDAKITemplate',
+    sortOrder: 1
+  },
+  {
+    question: 'Keep',
+    description: 'What should we continue doing?',
+    groupColor: '#D9D916',
+    templateId: 'dropAddKeepImproveDAKITemplate',
+    sortOrder: 2
+  },
+  {
+    question: 'Improve',
+    description: 'What should we improve or revise?',
+    groupColor: '#7373E5',
+    templateId: 'dropAddKeepImproveDAKITemplate',
+    sortOrder: 3
+  },
+  {
+    question: 'To Discuss',
+    description: `What should we talk about? We'll group related items into topics and vote on them to decide what to chat about`,
+    groupColor: '#45E5E5',
+    templateId: 'leanCoffeeTemplate',
+    sortOrder: 0
+  },
+  {
+    question: 'Keep Doing 🌀',
+    description: `What behaviors are working and adding value?`,
+    groupColor: '#7373E5',
+    templateId: 'starfishTemplate',
+    sortOrder: 0
+  },
+  {
+    question: 'Less of ➖',
+    description: `What are we overdoing or spending too much time on?`,
+    groupColor: '#D9D916',
+    templateId: 'starfishTemplate',
+    sortOrder: 1
+  },
+  {
+    question: 'More of ➕',
+    description: `What should we take advantage of or spend more time on?`,
+    groupColor: '#45E5E5',
+    templateId: 'starfishTemplate',
+    sortOrder: 2
+  },
+  {
+    question: 'Stop ⏹️',
+    description: `What isn't adding value or helping the team?`,
+    groupColor: '#E55C5C',
+    templateId: 'starfishTemplate',
+    sortOrder: 3
+  },
+  {
+    question: 'Start ⏯️',
+    description: `What new things should we explore or start doing?`,
+    groupColor: '#52CC52',
+    templateId: 'starfishTemplate',
+    sortOrder: 4
+  },
+  {
+    question: 'What went well 😄',
+    description: `What good things happened this sprint?`,
+    groupColor: '#7373E5',
+    templateId: 'whatWentWellTemplate',
+    sortOrder: 0
+  },
+  {
+    question: `What didn't go well 😞`,
+    description: `What didn't go as planned or desired this sprint?`,
+    groupColor: '#E55C5C',
+    templateId: 'whatWentWellTemplate',
+    sortOrder: 1
+  }
+] as const
+
+const templates = templateNames.map((templateName) => makeTemplate(templateName))
+const reflectPrompts = promptsInfo.map((promptInfo: PromptInfo) => makePrompt(promptInfo))
+
+export const up = async function(r: R) {
+  try {
+    await Promise.all([
+      r
+        .table('MeetingTemplate')
+        .insert(templates)
+        .run(),
+      r
+        .table('ReflectPrompt')
+        .insert(reflectPrompts)
+        .run(),
+      // we cache the templates but get the prompts from the db
+      // clear the cache to prevent empty templates
+      redis.flushall()
```

Comment:
I see the reasoning behind doing this in the down migration, but why do we need to flush on the up migration?

Reply:
If we don't do it in the up migration, `publicTemplates` will be read from the cache rather than the updated db, so it won't have the new retro templates. I could add the new template to Redis but I'd prefer to clear the key to make sure there isn't a mismatch between the db and Redis. I've updated the comment to make it clearer

Reply:
ahhhh i see, i assumed we were cacheing individual templates but from your code link i see we're cacheing the entire batch of retro templates. that makes sense!